lib/core/viewmodels/schedule_viewmodel.dart

```diff
@@ -87,11 +98,66 @@ class ScheduleViewModel extends FutureViewModel<List<CourseActivity>> {
     return events;
   }
 
+  void handleViewChanged(DateTime date, EventController controller) {
+    selectedDate = date;
+    final eventsToAdd = selectedWeekCalendarEvents();
+    final List<CalendarEventData> eventsAdded = [];
+    for (int i = 0; i < eventsToAdd.length; i++) {
+      if (!calendarEvents.contains(eventsToAdd[i])) {
+        calendarEvents.add(eventsToAdd[i]);
+        eventsAdded.add(eventsToAdd[i]);
+      }
+    }
+
+    controller.addAll(eventsAdded);
+  }
+
+  List<CalendarEventData> selectedDateCalendarEvents(DateTime date) {
+    final seen = <String>{};
+    return _coursesActivities[DateTime(date.year, date.month, date.day)]
+            ?.map((e) => CalendarEventData(
```

Comment:
Serait-il possible de mettre un nom significatif pour remplacer `e` ?

Reply:
Fixed!

Reply:
Not fixed `?.map((e) => CalendarEventData(`